slices/slices_test.go

```diff
@@ -0,0 +1,226 @@
+// Copyright 2023 The Authors (see AUTHORS file)
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package slices
+
+import (
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+)
+
+func TestMap(t *testing.T) {
+	t.Parallel()
```

Comment:
This Parallel call seems unnecessary if the subtests are being run in Parallel. I'm not sure what happens if Parallel is called twice in a goroutine/thread either.

Reply:
see https://github.com/abcxyz/readability/wiki/Go-Best-Practices#table-driven

Reply:
The first call to `t.Parallel()` causes `TestMap` to run in parallel with all the other tests marked `t.Parallel()`. The table-driven tests inside will run in serial to each other (but _potentially_ in parallel to other table-driven tests running in serial in a different top-level test).

Putting `t.Parallel()` on the subtests allows all subtests to run in parallel too. This effectively maximizes the number of tests running in parallel across all CPUs, and it's generally the way to get the fastest runtimes. This is especially true when the race detector is enabled.